client/data/help/use-active-support-tickets-query.js

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from 'react-query';
+import wpcom from 'calypso/lib/wp';
+
+const ACTIVE_STATUSES = [ 'New', 'Open', 'Hold' ];
+
+// Support History data needs to be disabled temporarily to avoid rate-limiting with our Zendesk API.
+// It is only used for non-essential read-only operations right now.
+const ENABLE_SUPPORT_HISTORY = false;
+
+export const useActiveSupportTicketsQuery = ( email, queryOptions = {} ) =>
+	useQuery(
+		[ 'activeSupportTickets', email ],
+		() => wpcom.req.get( '/support-history', { email, apiNamespace: 'wpcom/v2' } ),
+		{
+			enabled: !! email && ENABLE_SUPPORT_HISTORY,
+			select: ( { data } ) =>
+				data.filter(
+					( item ) => item.type === 'Zendesk_History' && ACTIVE_STATUSES.includes( item.status )
+				),
```

Comment:
One downside of using the `select` function, and selecting only `.length` from it in the HOC, is that the _entire_ unfiltered array is stored in persistent storage:
<img width="790" alt="Screenshot 2021-12-12 at 10 38 31" src="https://user-images.githubusercontent.com/664258/145707463-1188f85d-6354-4a78-a77a-aabb1e0c2a49.png">
To store only the count, the query function itself would have to do the filtering:
```js
() => wpcom.req.get().then(({data}) => data.filter().length
```
Or maybe disable persistence for this query completely? Not sure if it makes sense here.

Reply:
I think it makes sense to remove persistence here TBH. The support ticket can be closed at any time and having the number of open tickets as fresh as possible makes sense. I'll update that before merging.

Reply:
Removed persistence for that query in 42cede6987fb147dba98514eea2e4f8552452da5